src/foundry/client/documents/abstract/document-collection.d.mts

```diff
@@ -1,22 +1,41 @@
-import type Document from "#common/abstract/document.d.mts";
-import type Application from "#client/appv1/api/application-v1.d.mts";
-import type ApplicationV2 from "#client/applications/api/application.d.mts";
+import type { InexactPartial, Identity, DeepPartial, AnyObject } from "#utils";
+import type { Collection } from "#common/utils/_module.d.mts";
+import type { Document } from "#common/abstract/_module.d.mts";
+import type { Application } from "#client/appv1/api/_module.d.mts";
+import type { ApplicationV2 } from "#client/applications/api/_module.d.mts";
+import type { SearchFilter } from "#client/applications/ux/_module.d.mts";
+import type { DatabaseAction, DatabaseOperationMap, DatabaseUpdateOperation } from "#common/abstract/_types.mjs";
+
+/** @privateRemarks `DatabaseBackend` is only used for links */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type DatabaseBackend from "#common/abstract/backend.d.mts";
+
+/** @privateRemarks `CompendiumCollection` and `CompendiumFolderCollection` only used for links */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { CompendiumCollection, CompendiumFolderCollection } from "#client/documents/collections/_module.d.mts";
+
+/** @privateRemarks `WorldCollection` only used for links */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { WorldCollection } from "#client/documents/abstract/_module.d.mts";
 
 /**
  * An abstract subclass of the Collection container which defines a collection of Document instances.
+ * @remarks The `data` passed to the constructor is *not* put unconditionally into the `Collection`, it's set as
+ * {@linkcode DocumentCollection._source | #_source} and then {@linkcode DocumentCollection._initialize | #initialize}
+ * adds valid documents to the `Collection` and invalid `id`s to {@linkcode DocumentCollection.invalidDocumentIds | #invalidDocumentIds}.
  */
-declare class DocumentCollection<
-  DocumentType extends Document.Type,
-  Methods extends Collection.Methods.Any = DocumentCollection.Methods<DocumentType>,
-> extends foundry.utils.Collection<Document.StoredForName<DocumentType>, Methods> {
-  constructor(data: Document.SourceForName<DocumentType>[]);
+declare abstract class DocumentCollection<
+  DocumentName extends Document.Type,
+  Methods extends Collection.Methods.Any = DocumentCollection.Methods<DocumentName>,
+> extends Collection<Document.StoredForName<DocumentName>, Methods> {
+  constructor(data?: Document.CreateDataForName<DocumentName>[]);
 
   /**
    * The source data array from which the Documents in the WorldCollection are created
+   * @remarks Created via `Object.defineProperty` during construction with `{ writable: false }`,
+   * and the `data` param from the constructor as `value`.
    */
-  _source: Document.SourceForName<DocumentType>[];
+  readonly _source: Document.CreateDataForName<DocumentName>[];
```

Comment:
I recall you talking about this in Discord but the only calls all use `fromSource` which makes me feel a bit conflicted. I doubt anyone will be mad it's wider than they expected though...